### PR TITLE
Handlebars: fix formatting of attributes

### LINKED ISSUE #8648: Handlebars: newline in an attribute incorrectly rendered as `[object Object]`
As reported in https://github.com/jgwhite/prettier/issues/1#issuecomment-640699531

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAYhCACAAgQwCcBzAXgB0RypKB6APhABoQIAHGAS2gGdlQihCAHcACkQR8U+ADbD8ATz4sARoXxgA1nBgBlNhs5RiyGIQCucFgAsYAWxkB1a53g8DYOLsmvOAN1cFZHAeZRAjHjhCGFF1Yjt8ZAAzWUiWACseAA8AIXUtHV18OzgAGSM4ZNSrEEys3SNiGTgARXMIeCqZNJADQkjCYKbOOxLBljZCIxhHTgATGGtkAA4ABgmhSMd1NmDJuAG-SpYAR3b4WPYpEHweAFooODg55+YQQjgzzg-Y-HjEpApbo1SJ2TimCwgxrNNodSqA6osGD4FSzBZLJAAJiR6k4MkaAGEIKMASADgBWN7mSIAFRRUiBPT8lgAklAXrBdGAphwAILs3QwBTNLqRAC+YqAA)

**Input:**

```
<Foo arg="
"/>
```

**Output:**

```
<Foo arg="[object Object]" />
```

**Expected behavior:**

```
<Foo arg="
" />
```

### PR DESCRIPTION
**Pull request by thorn0**
## Description

This PR
* fixes escaping of `{{` in attributes and text
* fixes the choice between `'` and `"` for attributes with interpolations
* fixes the bug with `[object Object]` printed in `class`
* implements simple formatting for `class`, like in HTML in the current stable version (before #7865)

Closes #8648
Closes #8667

cc @dcyriller @kangax 

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/printer-glimmer.js [hunk 1]
```diff
@@ -674,6 +685,11 @@ function generateHardlines(number = 0, max = 0) {
  * @param {object} options - the prettier options object
  */
 function printStringLiteral(stringLiteral, options) {
+  const { quote, regex } = chooseEnclosingQuote(options, stringLiteral);
+  return [quote, stringLiteral.replace(regex, `\\${quote}`), quote];
+}
+
+function chooseEnclosingQuote(options, stringLiteral) {
```
Comment (dcyriller) [inline src/language-handlebars/printer-glimmer.js:692]: 👍

File: changelog_unreleased/handlebars/10145.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#### Fix formatting of attributes (#10145 by @thorn0)
+
+- fix escaping of `{{` in attributes and text
```
Comment (dcyriller) [inline changelog_unreleased/handlebars/10145.md:3]: nice catch, I had this refactor / fix on a branch too

**Review by dcyriller — APPROVED** [review]
Left one small comment about `n.chars` that might be transformed into `text`. Sounds like a great improvement overall.

Do you think we should use `src/language-html/syntax-attribute.js` functions to format handlebars attributes in the future?

**Comment by thorn0** [comment]
@dcyriller 

> Do you think we should use src/language-html/syntax-attribute.js functions to format handlebars attributes in the future?

What can be reused should definitely be reused, but we need to figure out a way to share that code between the two printers. Currently `language-html` and `language-handlebars` are built as separate bundles.

**Comment by thorn0** [comment]
@dcyriller Just realized that I was wrong here:

> Currently language-html and language-handlebars are built as separate bundles

Only the parsers are extracted to separate bundles. The printers all go in one bundle, so it's okay to `require` things directly from `language-html` in the printer.

**Pull request merged**